### ISSUE TITLE
Is ESP32 BLE Advertiser unconnectable status means it's unscannable?

### ISSUE THREAD

**Issue by Walkline80** [micropython/micropython#5738]
I use 2 esp32 boards as Advertiser and Scanner, here is my sample codes:

Advertiser part:
```python
import ubluetooth as bt

def main():
    ble = bt.BLE()

    ble.active(False)
    ble.active(True)

    # iBeacon payload
    adv_payload = bytearray(b"\x02\x01\x06\x1a\xffL\x00\x02\x15MyA\x9f\xb1\x80O\xf6\x8c\xb6\x9f\xa1\xb5\x7f\xb1h'\x1a'f\xc7")

    ble.gap_advertise(None)
    ble.gap_advertise(100000, adv_data=adv_payload, connectable=False)

if __name__ == "__main__":
    main()

>>> GAP procedure initiated: advertise; disc_mode=2 adv_channel_map=7 own_addr_type=0 adv_filter_policy=0 adv_itvl_min=160 adv_itvl_max=160
```

Scanner part:
```python
import ubluetooth as bt

def main():
    def irq(event, data):
        if event == 1 << 4: # scan result
            print(data)

    ble = bt.BLE()

    ble.active(False)
    ble.active(True)

    ble.irq(irq)

    ble.gap_scan(None)
    ble.gap_scan(0, 1000000, 100000)

if __name__ == "__main__":
    main()

>>> GAP procedure initiated: discovery; own_addr_type=0 filter_policy=0 passive=1 limited=0 filter_duplicates=0 duration=forever
```

While both boards runs correctly, the Scanner can not find the Advertiser's broadcast, but running `nRF Connect` with my mobile can find it.

I've tried turn the Advertiser's `connectable` param to `True`, and then the Scanner can find it easily.

Is there any solution for this issue?

**Comment by dpgeorge** [comment]
> While both boards runs correctly, the Scanner can not find the Advertiser's broadcast, but running `nRF Connect` with my mobile can find it.
> 
> I've tried turn the Advertiser's `connectable` param to `True`, and then the Scanner can find it easily.
> 
> Is there any solution for this issue?

Yes I've noticed this bug as well.  It's because `extmod/nimble/modbluetooth_nimble.c:gap_scan_cb()` is not handling all incoming scan events.

**Comment by dpgeorge** [comment]
I posted a fix in #5743.

**Comment by dpgeorge** [comment]
Fixed by dd0bc26e65734b8a4fafa3769008e92e2ec6645d
